### DISCUSSION TITLE
Igx-Grid how to have cell editor as combobox and still have filters on that column

### DISCUSSION THREAD

**Discussion by KaushikEDS** [IgniteUI/igniteui-angular#12148] [Q&A]
We are facing issues with igx-grid component having cell editable. 
In igxCellEditor template, I have an igx-combo component. The code for the same given below 
`                                <ng-template  igxCellEditor let-cell="cell">
                                    <igx-combo #dropdownCombo1 type="line"  [itemsMaxHeight]="250" [data]="myData" [igxFocus]="true"
                                 [displayKey]="'valueKey'" [valueKey]="'optionValue'" [(ngModel)]="cell.editValue"></igx-combo>
                                </ng-template>`
                                
I have implemented this in my grid and am facing issues with filtering.
When I reapply the filters using my state the grid.filteringExpressionTree has the proper value but the grid isn't filtered at all.

Can I have a working example of a cell editor with igx-combo filtering when we reapply filters on that column?

**Comment by amitmoondra** [comment]
Hi @hanastasov ,
This happens when a column has cell values as [] => array, and the saved filters when reapplied doesnot work, and the grid is not filtered correctly.
Please can you get this checked ?

**Comment by hanastasov** [comment]
hi @KaushikEDS  and @amitmoondra ,

I was not able to reproduce this behavior on my end using the following steps:

1) Run a grid with StatePersistence directive - [SAMPLE LINK](https://stackblitz.com/edit/angular-qug4i6?file=src/app/grid/grid-save-state/grid-state.component.html)
2) Add a combo editor for a certain column
```html
                        <ng-template igxCellEditor let-cell="cell" >
                            <igx-combo #dropdownCombo1 type="line"
                                [itemsMaxHeight]="250"
                                [data]="localData"
                                [igxFocus]="true"
                                [displayKey]="'Country'"
                                [valueKey]="'EmployeeID'"
                                [(ngModel)]="cell.editValue">
                            </igx-combo>
                        </ng-template>
```
3) Apply filtering on the grid (both the column with the combo editor and/or another one)
4) Save the grid state
5) Clear the grid state
6) Reapply the grid state

Result:
Same filter as in step 3) is applied on the grid.

Could you please update the sample to reproduce the behavior? Or provide another sample where I can investigate it.
